### PR TITLE
Created ValidationCallback

### LINKED ISSUE #1628: Optional logging of validation loss (and other metrics) in KerasModel
KerasModel (and TensorGraph) currently don't support periodic logging of validation loss.
Would it be a good idea to have this in the `fit_generator` and `fit` API?

This would be an optional argument with the modified API looking something like:

```
def fit(self, dataset, nb_epoch=10, max_checkpoints_to_keep=5, 
      checkpoint_interval=1000, deterministic=False, restore=False, 
      submodel=None, val_dataset=None, eval_interval=1000, **kwargs):
```

It would make it easier to do things like early stopping, which was done in the ChemNet Transfer Learning paper, and in general can be useful as well

### PR DESCRIPTION
**Pull request by peastman**
Fixes #1628.

### CODE REVIEW

**Comment by peastman** [comment]
Another of these cases.  On travis, one test consistently fails.  But I've tested repeatedly on two different computers, and I just can't get it to fail.  It never even comes close.  Can someone else try and see if you can reproduce the failure?

**Comment by peastman** [comment]
Ok, I've finally got the tests passing!  (Other than the MAML test that we know fails on Python 2.)  Any comments on the implementation would be great.

**Comment by peastman** [comment]
@rbharath do you have any thoughts on this?

**Comment by rbharath** [comment]
Feel free to merge whenever ready :)

**Comment by peastman** [comment]
Thanks!
